### DISCUSSION TITLE
Decimals as custom field data types

### DISCUSSION THREAD

**Discussion by eronlloyd** [netbox-community/netbox#10318] [Ideas]
Now that 3.3 is out, I'm exploring ways to effectively document best practices for fiber terminations in Netbox since #9102 has been addressed. One of the first things I'm trying to do is create custom fields for insertion loss (IL) for ports and attenuation for cables, but this needs to be in decimal format to accurately represent the decibel values.

Would adding a Decimal type to custom fields be difficult to enable this use case?

**Comment by jcollie** [comment]
As a stopgap you could create the fields as text fields and add a validation regex:

```
^[+-]?[0-9]+(\.[0-9]+)?$
```

**Reply by eronlloyd** [reply]
Yeah, that's exactly what I was thinking. The consideration then would have to be what needs to happen in type conversion. I see that you can change a custom field's type, but I'm not sure yet what Netbox does with existing values.

For the full implementation, I also wonder if it would make sense to also allow Floats, too, as a lot of the code needed for each choice type would be reusable. It's probably not a big deal, but Floats are less expensive in overhead and good enough in my case.

**Comment by jeremystretch** [comment]
> Would adding a Decimal type to custom fields be difficult to enable this use case?

Should be doable. One potential issue is that JSON doesn't natively distinguish between integer and decimal values (it just knows "numbers"), however that logic should be handled by the custom field's serialization methods anyway.

@eronlloyd would you like to submit a feature request for this?

**Reply by eronlloyd** [reply]
Absolutely, @jeremystretch. Will get that in tonight, thanks!
